### PR TITLE
Add split gem and an example A/B test, resolves #411

### LINKED ISSUE #411: Add split gem
https://github.com/splitrb/split

This gem will allow us to do self hosted A/B testing.

Please check to make sure the management console is protected by at least basic auth

### PR DESCRIPTION
**Pull request by hollomancer**
This is for review, we can probably select a better A/B test than this.

Also needs to point to user auth with an env variable.

### CODE REVIEW

**Comment by hollomancer** [comment]
Maybe A/B testing #429 will be better.

File: config/initializers/split.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Split::Dashboard.use Rack::Auth::Basic do |username, password|
+  username == 'test' && password == 'test'
```
Comment (rickr) [inline config/initializers/split.rb]: Can you set these to environment variables? Maybe something like split_username and split_password?

**Comment by rickr** [comment]
Looks good! Is this test shipping live or do we have to start it?

**Comment by rickr** [comment]
Oh - one other thing. The test starts on the home view and ends on the donations controller. Is that intended?

**Comment by hollomancer** [comment]
This is just a test to see if it's working, and yeah, it's live. We probably want to do a different test that has more of a user acquisition impact.

**Comment by rickr** [comment]
So the test will only finish if a user hits the home page, then goes to the donate page? Would it make more sense to finish the test to the sign up controller?

**Comment by hollomancer** [comment]
Sure, we can do that. We should start coming up with more tests to do, and make them beginner friendly issues.

**Pull request merged**